docs/2022-04-07-Full-Corectness-All-Path-Reachability-Proofs.md

```diff
@@ -0,0 +1,412 @@
+Proving Full Correctness All-Path Reachability Claims
+=====================================================
+
+This document details Full Correctness All-Path Reachability without solving the
+most-general-unifier (MGU) problem.
+MGU will be detailed in a separate document.
+
+_Prepared by Traian Șerbănuță, based on [proving All-Path Reachability
+claims](2019-03-28-All-Path-Reachability-Proofs.md)_
+
+Definitions
+-----------
+
+### All-path finally
+
+Given a formula `φ`, let `♢φ` denote the formula “all-path finally” `φ`,
+defined by:
+
+```
+♢φ := μX.φ ∨ (○X ∧ •⊤)
```

Comment:
I suggest we use a different symbol than ♢. The symbol ♢ is usually used to mean (one-path) eventually, which is defined as
```
♢φ := μX . φ \/ •X
```
It's also different from the encoding of all-path reachability, which uses the greatest fixpoint (but the rest of the pattern is the same).

So, is it a new constructs that differs from all-path reachability only in that it uses the least fixpoint? What's the semantics of this new construct? Could you explain it in the model?

Reply:
I've changed it to `AF` (all-paths finally), since it's basically the "total" version of `wAF`.  I've also detailed more the semantics (model interpretation) of the operator.